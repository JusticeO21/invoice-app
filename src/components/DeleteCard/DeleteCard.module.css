.container{
    background-color: var(--secondary-bg);
    padding: 3rem;
    border-radius: 0.35rem;
    box-shadow: var(--box-shadow);
    display: grid;
    gap: 1.5rem;
}

.actions{
    width: 100%;
    display: flex;
    justify-content: end;
    gap: 2rem;
}

.actions button {
    padding: 1rem 2rem;
}
