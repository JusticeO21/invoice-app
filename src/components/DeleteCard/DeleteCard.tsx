import React from "react";
import styles from "./DeleteCard.module.css";
import { Heading } from "../heading/Heading";
import { Text } from "../text/Text";
import Button from "../button/Button";

type DeleteProps = {
  invoiceId: string;
  onDelete: () => void;
  onCancel: () => void;
  isLoading?: boolean;
};

function DeleteCard({ invoiceId, onDelete, onCancel, isLoading }: DeleteProps) {
  function handleDelete(event: React.MouseEvent) {
    event.stopPropagation();
    onDelete();
  }

  function handleCancel(event: React.MouseEvent) {
    event.stopPropagation();
    onCancel();
  }

  return (
    <div className={styles.container}>
      <Heading>Confirm Deletion</Heading>
      <Text>
        {`Are you sure you want to delete invoice #${invoiceId}? This action cannot be
        undone.`}
      </Text>

      <div className={styles.actions}>
        <Button variant="secondary" onClick={handleCancel} disabled={isLoading}>
          Cancel
        </Button>
        <Button variant="danger" onClick={handleDelete} disabled={isLoading}>
          {isLoading ? "...Loading" : "Confirm"}
        </Button>
      </div>
    </div>
  );
}

export default DeleteCard;
