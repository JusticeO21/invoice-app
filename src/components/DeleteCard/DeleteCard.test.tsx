
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import DeleteCard from "./DeleteCard";

describe("DeleteCard Component", () => {
  it("should render the component with the correct content", () => {
    const invoiceId = "12345";
    render(
      <DeleteCard invoiceId={invoiceId} onDelete={vi.fn()} onCancel={vi.fn()} />
    );

    const heading = screen.getByRole("heading", { name: "Confirm Deletion" });
    expect(heading).toBeInTheDocument();

    const text = screen.getByText(
      `Are you sure you want to delete invoice #${invoiceId}? This action cannot be undone.`
    );
    expect(text).toBeInTheDocument();

    const cancelButton = screen.getByRole("button", { name: /cancel/i });
    expect(cancelButton).toBeInTheDocument();

    const confirmButton = screen.getByRole("button", { name: /confirm/i });
    expect(confirmButton).toBeInTheDocument();
  });

  it("should call onCancel when Cancel button is clicked", () => {
    const onCancel = vi.fn();
    const invoiceId = "12345";
    render(
      <DeleteCard
        invoiceId={invoiceId}
        onDelete={vi.fn()}
        onCancel={onCancel}
      />
    );

    const cancelButton = screen.getByRole("button", { name: /cancel/i });
    fireEvent.click(cancelButton);

    expect(onCancel).toHaveBeenCalledTimes(1);
  });

  it("should call onDelete when Confirm button is clicked", () => {
    const onDelete = vi.fn();
    const invoiceId = "12345";
    render(
      <DeleteCard
        invoiceId={invoiceId}
        onDelete={onDelete}
        onCancel={vi.fn()}
      />
    );

    const confirmButton = screen.getByRole("button", { name: /confirm/i });
    fireEvent.click(confirmButton);

    expect(onDelete).toHaveBeenCalledTimes(1);
  });

  it("should render correctly when the invoiceId is dynamic", () => {
    const invoiceId = "67890";
    render(
      <DeleteCard invoiceId={invoiceId} onDelete={vi.fn()} onCancel={vi.fn()} />
    );

    const text = screen.getByText(
      `Are you sure you want to delete invoice #${invoiceId}? This action cannot be undone.`
    );
    expect(text).toBeInTheDocument();
  });

  it("should pass correct classes for the buttons", () => {
    const invoiceId = "12345";
    render(
      <DeleteCard invoiceId={invoiceId} onDelete={vi.fn()} onCancel={vi.fn()} />
    );

    const cancelButton = screen.getByRole("button", { name: /cancel/i });
    expect(cancelButton.className).toContain("secondary");

    const confirmButton = screen.getByRole("button", { name: /confirm/i });
    expect(confirmButton.className).toContain("danger");
  });

  it("should match snapshot", () => {
    const invoiceId = "12345";
    const { asFragment } = render(
      <DeleteCard invoiceId={invoiceId} onDelete={vi.fn()} onCancel={vi.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
