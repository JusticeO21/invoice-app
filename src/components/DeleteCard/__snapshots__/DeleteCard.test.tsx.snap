// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeleteCard Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_2136ee"
  >
    <h1
      class="_heading_daf655 _h1_daf655"
    >
      Confirm Deletion
    </h1>
    <p
      class="text description undefined"
    >
      Are you sure you want to delete invoice #12345? This action cannot be
        undone.
    </p>
    <div
      class="_actions_2136ee"
    >
      <button
        class="_button_4fb095 _secondary_4fb095 _rounded_4fb095"
      >
        Cancel
      </button>
      <button
        class="_button_4fb095 _danger_4fb095 _rounded_4fb095"
      >
        Confirm
      </button>
    </div>
  </div>
</DocumentFragment>
`;
